File: C++/find-kth-character-in-expanded-string.cpp
// Time:  O(n)
// Space: O(1)

// string
class Solution {
public:
    char kthCharacter(string s, long long k) {
        int i = 0;
        for (int l = 0; i < size(s); ++i) {
            if (s[i] == ' ') {
                l = 0;
                --k;
            } else {
                k -= ++l;
            }
            if (k < 0) {
                break;
            }
        }
        return s[i];
    }
};

File: Python/find-kth-character-in-expanded-string.py
# Time:  O(n)
# Space: O(1)

# string
class Solution(object):
    def kthCharacter(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: str
        """
        l = 0
        for i in xrange(len(s)):
            if s[i] == ' ':
                l = 0
                k -= 1
            else:
                l += 1
                k -= l
            if k < 0:
                break
        return s[i]
